### PR TITLE
Better events validation / better names and docs

### PR DESCRIPTION
**Pull request by jpslav**
* Previously specific event validation was not happening.
* Changes the name of the /events POST body to `payload`
* Returns error messages for events in the body.
* Adds tests

### CODE REVIEW

File: app/controllers/api/v0/events_controller.rb [hunk 1]
```diff
@@ -33,8 +33,11 @@ def convert_api_data_to_kafka_data(api_data:)
       # Adjust the client's occurred at time by a device sent at adjustment
       data[:occurred_at] = TimeUtil.infer_actual_occurred_at_from_client_timestamps(
         request_received_at: request.env[:received_at],
-        client_clock_occurred_at: api_data[:client_clock_occurred_at],
-        client_clock_sent_at: api_data[:client_clock_sent_at]).to_i
+        client_clock_occurred_at: data[:client_clock_occurred_at],
+        client_clock_sent_at: data[:client_clock_sent_at]
+      ).to_i
+
+      data.except!(:client_clock_sent_at, :client_clock_occurred_at)
```
Comment (steveburkett) [inline app/controllers/api/v0/events_controller.rb:40]: this should be tested...that we send a client_x data field but just get back a occurred_at field.

File: lib/v0_bindings_extensions.rb [hunk 2]
```diff
@@ -6,11 +6,35 @@
...
+
+  Api::V0::Bindings::Events.class_exec do
+    alias_method :original_valid?, :valid?
+    def valid?
+      original_valid? && events.all?(&:valid?)
+    end
+
+    alias_method :original_list_invalid_properties, :list_invalid_properties
+    def list_invalid_properties
+      event_invalid_properties = events.map.with_index do |event,ii|
```
Comment (steveburkett) [inline lib/v0_bindings_extensions.rb]: suggestion..to not use cryptic abbreviations like i, x, y, ii, etc.  and to say what it really is.  index.

**Review by steveburkett — APPROVED** [review]

File: app/controllers/api/v0/events_controller.rb (hunk 1 above)
Comment (jpslav) [reply app/controllers/api/v0/events_controller.rb:40]: added spec

File: lib/v0_bindings_extensions.rb (hunk 2 above)
Comment (jpslav) [reply lib/v0_bindings_extensions.rb]: fixed.  old habits :-)  (from when I was annoyed with people using just a single letter `i` :-)

**Pull request merged**